shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java

```diff
@@ -0,0 +1,342 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.loader;
+
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.content.res.AssetManager;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.os.SystemClock;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+import android.view.WindowManager;
+
+import io.flutter.BuildConfig;
+import io.flutter.embedding.engine.FlutterJNI;
+import io.flutter.util.PathUtils;
+import io.flutter.view.VsyncWaiter;
+
+import java.io.File;
+import java.util.*;
+
+/**
+ * This class finds Flutter resources in an application APK and also loads Flutter's native library.
+ */
+public class FlutterLoader {
+    private static final String TAG = "FlutterLoader";
+
+    // Must match values in flutter::switches
+    private static final String AOT_SHARED_LIBRARY_NAME = "aot-shared-library-name";
+    private static final String SNAPSHOT_ASSET_PATH_KEY = "snapshot-asset-path";
+    private static final String VM_SNAPSHOT_DATA_KEY = "vm-snapshot-data";
+    private static final String ISOLATE_SNAPSHOT_DATA_KEY = "isolate-snapshot-data";
+    private static final String FLUTTER_ASSETS_DIR_KEY = "flutter-assets-dir";
+
+    // XML Attribute keys supported in AndroidManifest.xml
+    public static final String PUBLIC_AOT_SHARED_LIBRARY_NAME =
```

Comment:
nit: If these are public they should also have Javadoc. It looks like they may be able to be private or package private though?

https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required

Reply:
Made private.